Appserver/Controllers/TimesheetController.cs

```diff
@@ -121,22 +121,53 @@ public IActionResult Submit(int id)
                 childform.FromJson(a);
                 textractform.AddPages(childform);
             }
-            var tsf = (TimesheetForm)AbstractFormObject.FromTextract(textractform);
-
-            tsf.id = id;
+            var ts = (TimesheetForm)AbstractFormObject.FromTextract(textractform);
 
-            var dbutil = new FormToDbUtil(_subcontext, _context);
-            Timesheet ts = dbutil.PopulateTimesheet(tsf);
-            dbutil.PopulateTimesheetEntries(tsf, ts);
+            ts.id = id;
+            Func<string, PWAsubmissionVals> PWAConv = (x) =>
+            {
+                var val = new PWAsubmissionVals();
+                val.value = x;
+                val.wasEdited = false;
+                return val;
+            };
+
+            var PWAForm = new PWAsubmission();
+            PWAForm.approval = PWAConv(ts.approval.ToString());
+            PWAForm.authorization = PWAConv(ts.authorization.ToString());

```

Comment:
Authorization (the checkbox at the very end of the form) is a Boolean, not a string.

Reply:
Approval (the provider initials) _is_ a string, so approval's type is correct here.